### PR TITLE
Tests for OID and for status

### PR DESCRIPTION
**Pull request by kshyatt**
OID was missing an error throw and status's getindex wasn't really
tested.

### CODE REVIEW

File: test/libgit2.jl [hunk 1]
```diff
@@ -346,6 +348,7 @@ mktempdir() do dir
                 LibGit2.add!(repo, test_file)
                 status = LibGit2.GitStatus(repo)
                 @test length(status) != 0
+                @test status[0] == nothing
```
Comment (ararslan) [inline test/libgit2.jl]: I don't know how much it matters but I remember seeing Tony Kelman say somewhere that it's better to use `===` for `nothing`.

Comment (yuyichao) [inline test/libgit2.jl]: It's better since it's type independent. It matters less for tests though (but is still preferred).

Comment (ararslan) [inline test/libgit2.jl]: Good to know. Thanks, @yuyichao!

Comment (tkelman) [inline test/libgit2.jl]: what's the meaning of status returning an array with `nothing` in it?

Comment (kshyatt) [inline test/libgit2.jl]: Correct me if I'm wrong but I think it means there are no changes to any tracked files.

Comment (tkelman) [inline test/libgit2.jl]: does GitStatus not return any information about the staged added file?

**Comment by kshyatt** [comment]
Fixed my own foolishness and made the `BoundsError` more robust ... this isn't `unsafe_getindex` 😉

File: base/libgit2/status.jl [hunk 2]
```diff
@@ -14,12 +14,11 @@ function Base.length(status::GitStatus)
 end
 
 function Base.getindex(status::GitStatus, i::Csize_t)
-    if length(status) == 0
+    if i < 1 || i > length(status)
         throw(BoundsError())
     end
     entry_ptr = ccall((:git_status_byindex, :libgit2), Ptr{Void},
                        (Ptr{Void}, Csize_t), status.ptr, i-1)
-    entry_ptr == C_NULL && return nothing
```
Comment (tkelman) [inline base/libgit2/status.jl]: this should still probably do something other than segfault if it returns null

Comment (kshyatt) [inline base/libgit2/status.jl]: It only returns `C_NULL` if the index `i-1` is bad, which I already check for above.

Comment (tkelman) [inline base/libgit2/status.jl]: it would be more defensive to at least check and throw if something unexpectedly causes it to occur

**Comment by kshyatt** [comment]
@tkelman 64 bit linux failure seems to have little to do with me. This ok to merge?

**Pull request merged**